### ISSUE TITLE
`chunks_exact_to_as_chunks` still suggests invalid const generic parameter expression

### ISSUE THREAD

**Issue by adamreichold** [rust-lang/rust-clippy#17359]
### Summary

This is a follow-up to #17314 where `chunks_exact_to_as_chunks` still suggest an expression based on const generic parameters which stable Rust does not allow yet.

The main difference to #17314 appears to be that the source of the issue is a constant associated with a type parameter instead of a const generic parameter itself.

### Lint Name

chunks_exact_to_as_chunks

### Reproducer

I tried [this code](https://codeberg.org/adamreichold/robinson/src/commit/3a25c375b24fb86226a5b5fc4b8c6e677b71650f/src/tokenizer.rs#L574):

```rust
for chunk in text.as_bytes().chunks_exact(M::BYTES) {
```

I saw this happen:

```
warning: using `chunks_exact` with a constant chunk size
   --> src/tokenizer.rs:574:34
    |
574 |     for chunk in text.as_bytes().chunks_exact(M::BYTES) {
    |                                  ^^^^^^^^^^^^^^^^^^^^^^ help: consider using `as_chunks` instead: `as_chunks::<{ M::BYTES }>().0`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#chunks_exact_to_as_chunks
    = note: `#[warn(clippy::chunks_exact_to_as_chunks)]` on by default
```

I expected to see this happen:

The lint does not fire as applying the suggestion leads to

```
error: generic parameters may not be used in const operations
   --> src/tokenizer.rs:574:48
    |
574 |     for chunk in text.as_bytes().as_chunks::<{ M::BYTES }>().0 {
    |                                                ^ cannot perform const operation using `M`
    |
    = note: type parameters may not be used in const expressions
    = help: add `#![feature(generic_const_exprs)]` to allow generic const expressions
```

### Version

```text
rustc 1.98.0-nightly (f28ac764c 2026-06-23)
binary: rustc
commit-hash: f28ac764c36004fa6a6e098d15b4016a838c13c6
commit-date: 2026-06-23
host: x86_64-unknown-linux-gnu
release: 1.98.0-nightly
LLVM version: 22.1.7
```

Actually, I ran this using the master branch at 958b1b293efb22bfd0eb3dc58ba36ed94c3420c0 via `cargo dev setup toolchain`.

**Comment by steflon19** [comment]
i am still getting this clippy warning on `rustc 1.98.0 (88d9e12ae 2026-08-18)`

`chunks_exact(P::SIZE_BYTES)` suggested as should be `as_chunks::<P::SIZE_BYTES>()` , which does not compile with 

Where 
```rust
pub trait PixelBytes {
    const SIZE_BYTES: usize;
    ...
}
```
